internal/plugin/connectors/http/generic/oauth/v1/protocol_test.go

```diff
@@ -0,0 +1,352 @@
+package oauth1protocol
+
+import (
+	"io"
+	"io/ioutil"
+	gohttp "net/http"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func Test_extractOAuthParams(t *testing.T) {
+	paramMap := map[string]string{
+		consumerKey:          "consumerKey",
+		consumerSecret:       "consumerSecret",
+		token:                "token",
+		tokenSecret:          "tokenSecret",
+		oauthNonce:           "oauth_nonce",
+		oauthSignatureMethod: "oauth_signature_method",
+		oauthVersion:         "oauth_version",
+		oauthTimestamp:       "oauth_timestamp",
+	}
+	result := extractOAuthParams(paramMap)
+	// Nonce and TimeStamp are generated at runtime and unknown
+	result.TimeStamp = ""
+	result.Nonce = ""
+
+	want := oAuth1{
+		ConsumerKey:     "consumerKey",
+		ConsumerSecret:  "consumerSecret",
+		Nonce:           "",
+		SignatureMethod: "HMAC-SHA1",
+		TimeStamp:       "",
+		Token:           "token",
+		TokenSecret:     "tokenSecret",
+		Version:         "1.0",
+	}
+	if reflect.DeepEqual(result, want) {
+		t.Log("extractOAuthParams() Passed!")
+	} else {
+		t.Errorf("extractOAuthParams() FAILED, expected %v but got %v", want, result)
+	}
+}
+
+func createRequestBody(s string) io.ReadCloser {
+	return ioutil.NopCloser(strings.NewReader(s))
+}
+
+func Test_collectParameters(t *testing.T) {
+	type args struct {
+		oauth1 oAuth1
+		r      *gohttp.Request
+	}
+
+	params := args{
+		oauth1: oAuth1{
+			ConsumerKey:     "xvz1evFS4wEEPTGEFPHBog",
+			Nonce:           "kYjzVBB8Y0ZFabxSWbWovY3uYSQ2pTgmZeNu2VS4cg",
+			SignatureMethod: "HMAC-SHA1",
+			TimeStamp:       "1318622958",
+			Token:           "370773112-GmHxMAgYyLbNEtIKZeRNFsMKPR9EyMZeS9weJAEb",
+			Version:         "1.0",
+		},
+		r: &gohttp.Request{
+			Method: "POST",
+			URL: &url.URL{
+				Scheme:   "http",
+				Host:     "example.com",
+				Path:     "/wp-json/wp/v2/posts",
+				RawQuery: "include_entities=true",
+			},
+			Header: map[string][]string{
+				"Authorization": {"doesn't matter"},
+				"Content-Type":  {"application/x-www-form-urlencoded"},
+			},
+			Body: createRequestBody("status=Hello%20Ladies%20%2B%20Gentlemen%2C%20a%20signed%20OAuth%20request%21"),
```

Comment:
Nit: I'd change this (ladies and gentlemen) to be something more neutral, like "foo bar hello world".

Reply:
We pulled these examples straight from the Twitter examples, but that's a good point.